File: graphs/representation/EdgeList.cpp
#include <iostream>
#include <vector>
#include <utility> // for pair
using std::cout;
using std::endl;

class Graph{
private:
    std::vector<std::pair<int,int>> edgeList;
public:
    Graph(){
    }

    void addList(int src,int dst){
        this->edgeList.push_back({src,dst});
    }

    void printEL(){
        cout << "[";
        for(int i=0; i<this->edgeList.size(); i++){
            printf("(%d,%d)",this->edgeList[i].first, this->edgeList[i].second);
        }
        puts("]");
    }

};

int main(){
    Graph g;

    g.addList(1,2);
    g.addList(1,3);
    g.addList(2,3);

    g.printEL();
    return 0;
}

/*
https://leetcode.com/problems/redundant-connection/
*/

File: graphs/representation/EdgeList.py
from typing import List

class Graph:
    def __init__(self) -> None:
        self.edges = []

    def addEdge(self, src:int ,dst: int) -> None:
        self.edges.append([src,dst])

    def printGraph(self) -> None:
        print("[",end="")

        for (src,dst) in self.edges:
            print(f"[{src},{dst}]",end="")
        print("]")

    def getNode(self,index: int) -> List:
        return self.edges[index]


def main():
    g = Graph()

    g.addEdge(1,2)
    g.addEdge(1,3)
    g.addEdge(2,3)

    g.printGraph()
    print(g.getNode(0))

if __name__ == "__main__":
	main()


'''
graph picture:
https://leetcode.com/problems/redundant-connection/
'''